client/src/main/java/com/netflix/conductor/client/task/WorkflowTaskCoordinator.java

```diff
@@ -297,32 +305,43 @@ private void pollForTask(Worker worker) {
 			logger.error("Error when polling for task " + e.getMessage(), e);
 		}
 	}
-	
-	private void execute(Worker worker, Task task) {
+
+	private void handleException(Throwable t, TaskResult result, Worker worker, boolean updateTask, Task task) {
+		WorkflowTaskMetrics.executionException(worker.getTaskDefName(), t);
+		result.setStatus(TaskResult.Status.FAILED);
+		result.setReasonForIncompletion("Error while executing the task: " + t);
+		TaskExecLog execLog = result.getLog();
+		execLog.setError(t.getMessage());
+		for (StackTraceElement ste : t.getStackTrace()) {
```

Comment:
You're not capturing the cause here, which is the most important part. I would just capture the stack trace as a string.